README.md

```diff
@@ -1,60 +1,44 @@
 ## OkLayoutInflater
 
-AndroidX [AsyncLayoutInflater](https://developer.android.com/reference/androidx/asynclayoutinflater/view/AsyncLayoutInflater) has some limitations. Here is an improved version of AsyncLayoutInflater with coroutine.
+AndroidX [AsyncLayoutInflater](https://developer.android.com/reference/androidx/asynclayoutinflater/view/AsyncLayoutInflater)
+has some limitations. Here is an improved version of AsyncLayoutInflater with Coroutines.
 
 1. Single thread to do all the inflate work
-2. Inflatework is not lifecycle aware/There is no way to cancel ongoing inflation
+2. Inflate work is not lifecycle aware/There is no way to cancel ongoing inflation
 3. Does not support LayoutInflater.Factory2.
-4. The default size limit of the cache queue is 10. If it exceeds 10, it will cause the main thread to wait.
+4. The default size limit of the cache queue is 10. If it exceeds 10, it will cause the main thread
+   to wait.
 
 Using OkLayoutInflater, we have improved threading limitations.
 
-
 ### Usage
- **Example of usage in fragment**
+
+**Example in Fragment**
 
 ```
-private val okLayoutInflater by lazy { OkLayoutInflater(requireContext()) }
+private val okLayoutInflater by lazy { OkLayoutInflater(this) }
 
 override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
-        val v = inflater.inflate(R.layout.loader_view, container, false)
-
-        okLayoutInflater.inflate(
-            contentLayoutId,
-            container
-        ) {
-            (v as? ViewGroup)?.addView(it)
-        }
-        return v
-}
-
-override fun onDestroyView() {
-    okLayoutInflater.cancel()
-    super.onDestroyView()
+    val loadingView = inflater.inflate(R.layout.loader_view, container, false)
+    okLayoutInflater.inflate(contentLayoutId, container) {
+        (loadingView as? ViewGroup)?.addView(it)
+    }
+    return loadingView
 }
 ```
 
- **Example of usage in view**
+**Example of usage in View**
 
 ```
 private val okLayoutInflater by lazy { OkLayoutInflater(context) }
 
 override fun onAttachedToWindow() {
     super.onAttachedToWindow()
-    LayoutInflater.from(context).inflate(R.layout.viewstub_customer_tx_view, this, true)
-    okLayoutInflater.inflate(
-        merchant.okcredit.accounting.R.layout.transaction_view,
-        this
-    ) {
+    okLayoutInflater.inflate(R.layout.transaction_view, this) { inflatedView ->
         removeAllViews()
-        addView(it, LayoutParams.MATCH_PARENT)
+        addView(inflatedView, LayoutParams.MATCH_PARENT)
     }
 }
```

Comment:
Can auto cancellation be used in custom views?

Reply:
If the `Context` provided has a `Lifecycle` then Yes, auto cancellation will work just fine.
In case of a Fragment, `viewLifecycleOwner` will be used.

Also remember, I've added a log as well if the `Context` is not a `LifecycleOwner`

Reply:
I've added support for inflation via Custom View as well.
The inflation will be cancelled if the view is detached from the window.
See #[db8ae57](https://github.com/okcredit/OkLayoutInflater/pull/3/commits/db8ae57be1a5d48bdaf33f91912e72b5a0bf2514)